dbclient/ClustersClient.py

```diff
@@ -277,6 +281,77 @@ def import_cluster_configs(self, log_file='clusters.log', acl_log_file='acl_clus
                 resp = self.put(api, acl_args)
                 print(resp)
 
+    def _log_cluster_ids_and_original_creators(
+            self,
+            cluster_log_file,
+            user_name_to_user_id_log_file='user_name_to_user_id.log',
+            creators_file='original_creator_user_ids.log',
+            cluster_ids_file='cluster_ids_to_change_creator.log'):
+        """
+        Log the cluster_ids_to_change_creator and original_creator_user_ids
+        These can be used to edit the clusters to be owned by the correct/original creator instead of the
+        PAT token owner. Fails if user_name_to_user_id.log does not exist.
+
+        :param user_name_to_user_id_log_file: file that contains the userName to userId mapping of the DST workspace.
+                                              This file is exported as part of the users-import step.
+        :param cluster_log_file: file that contains the exported cluster objects.
+        :param creators_file: output file written with the list of original creators of clusters.
+                              The list should be in the same order as the cluster_ids_file.
+        :param cluster_ids_file: output file written with the list of cluster ids that need the creator tag to change.
+        """
+        cluster_ids_to_change_creator = []
+        original_creator_user_ids = []
+        with open(self.get_export_dir() + user_name_to_user_id_log_file, 'r') as fp:
+            user_name_to_user_id = json.loads(fp.read())
+
+        old_to_new_cluster_mapping = self.get_cluster_id_mapping(cluster_log_file)
+
+        with open(self.get_export_dir() + cluster_log_file, 'r') as fp:
+            for line in fp:
+                cluster_conf = json.loads(line)
+                original_cluster_creator = cluster_conf['creator_user_name']
+                original_cluster_id = cluster_conf['cluster_id']
+                if original_cluster_id in old_to_new_cluster_mapping and original_cluster_creator in user_name_to_user_id:
+                    current_cluster_id = old_to_new_cluster_mapping[original_cluster_id]
+                    cluster_ids_to_change_creator.append(current_cluster_id)
+                    original_creator_user_ids.append(user_name_to_user_id[original_cluster_creator])
+                else:
+                    print("The old cluster id " + original_cluster_id +
```

Comment:
Nit: we should also log original_cluster_creator in the error message.

Reply:
√